src/lease-manager.ts

```diff
@@ -257,7 +257,15 @@ export class LeaseManager extends EventEmitter {
       const lifespan = (Date.now() - message.received) / (60 * 1000);
 
       if (lifespan < this._options.maxExtensionMinutes!) {
-        message.modAck(deadline);
+        if (this._subscriber.isExactlyOnceDelivery) {
+          message.modAckWithResponse(deadline).catch(() => {
+            // In the case of a permanent failure (temporary failures are retried),
+            // we need to stop trying to lease-manage the message.
+            this.remove(message);
```

Comment:
We should be setting the response for the eventual ack as failed as well. This way, when the client acks the message, we do not have to issue a server call. If the modack fails with a permanent error, ack will also fail with a permanent error.

Reply:
@maheshgattani Can you check the change I just made to validate that it meets the requirements? The ack/modAck/nack methods will only throw a failure if the error is permanent, so I think this covers it.